bluesky/run_engine.py

```diff
@@ -1668,6 +1681,9 @@ def _checkpoint(self, msg):
 
     @asyncio.coroutine
     def _reset_checkpoint_state(self):
+        self._reset_checkpoint_state_meth()
+
+    def _reset_checkpoint_state_meth(self):
```

Comment:
Could you make this the main method, then just wrap with `asyncio.coroutine` to create the other one?
Bike-sheddy, but it feels a bit more natural to me:

```python
    def _reset_checkpoint_state(self):
        if self._msg_cache is None:
            return
        ...

    _reset_checkpoint_state_coroutine = asyncio.coroutine(_reset_checkpoint_state)
```

Almost seems like the coroutines should have a common prefix or something instead... but I've exceeded my picky comment limit, so ignore this.